packages/mini-browser/src/node/mini-browser-endpoint.ts

```diff
@@ -40,17 +40,39 @@ export const MiniBrowserEndpointHandler = Symbol('MiniBrowserEndpointHandler');
 export interface MiniBrowserEndpointHandler {
 
     /**
-     * Returns or resolves to a positive number if the current contribution can handle the resource with the given URI.
-     * The number indicates the priority of the endpoint contribution. If it is not a positive number, it means, the
-     * contribution cannot handle the URI.
+     * Returns with the file extensions supported by the current `mini-browser` endpoint handler.
+     * The file extension must start with the leading `.` (dot). For instance; `'.html'` or `['.jpg', '.jpeg']`.
+     * Extensions are case insensitive.
      */
-    canHandle(uri: string): MaybePromise<number>;
+    supportedExtensions(): string | string[];
 
     /**
      * Responds back to the sender.
      */
     respond(statWithContent: FileStatWithContent, response: Response): MaybePromise<Response>;
 
+    /**
+     * Returns or resolves to a positive number if the current contribution can handle the resource with the given URI.
+     * The number indicates the priority of the endpoint contribution. If it is not a positive number, it means, the
+     * contribution cannot handle the URI.
+     */
+    priority(uri: string): MaybePromise<number>;
```

Comment:
Given the current implementation, this should be done on the client side, too.

Reply:
This is exactly what we did and that caused the context menu issue because we could not run `async` code. Right now, we just assume if the extension can be handled, then it should [have a bigger priority than the `Code Editor`](https://github.com/theia-ide/theia/pull/1998/files#diff-efa0e82536aa80ae220e472ef9d04256R64).